software/py/vanilla_stats_parser.py

```diff
@@ -0,0 +1,656 @@
+#
+#   vanilla_stats_parser.py
+#
+#   vanilla core stats extractor
+# 
+#   input: vanilla_stats.log
+#   output: manycore_stats.log
+#   output: tile_stats/tile_#_stats.log for all tiles 
+#
+#   @author Borna
+#
+#   How to use:
+#   python3 generate_stats.py --dim-y {manycore_dim_y}  --dim-x {manycore_dim_x} 
+#                             --tile (optional) --tile_group (optional)
+#                             --input {vanilla_stats.log}
+#
+#   ex) python3 --input generate_stats.py --dim-y 4 --dim-x 4 --tile --tile_group --input vanilla_stats.log
+#
+#   {manycore_dim_y}  Mesh Y dimension of manycore 
+#   {manycore_dim_x}  Mesh X dimension of manycore 
+#   {per_tile}        Generate separate stats file for each tile default = False
+#   {input}           Vanilla stats input file     default = vanilla_stats.log
+
+
+
+import sys
+import argparse
+import os
+import re
+import csv
+from enum import Enum
+from collections import Counter
+
+
+
+# Default coordinates of origin tile
+BSG_ORG_X = 0
+BSG_ORG_Y = 1
+
+# Default input values
+DEFAULT_INPUT_FILE = "vanilla_stats.csv"
+
+
+
+
+
+# These values are used by the manycore library in bsg_print_stat instructions
+# they are added to the tag value to determine the tile group that triggered the stat
+# and also the type of stat (stand-alone stat, start, or end)
+# the value of these paramters should match their counterpart inside 
+# bsg_manycore/software/bsg_manycore_lib/bsg_manycore.h
+# For formatting, see the CudaStatTag class
+BSG_STAT_TAG_WIDTH   = 8
+BSG_STAT_TG_ID_WIDTH = 10
+BSG_STAT_X_WIDTH     = 6
+BSG_STAT_Y_WIDTH     = 6
+BSG_STAT_TYPE_WIDTH  = 2
+BSG_STAT_TAG_MASK   = ((1 << BSG_STAT_TAG_WIDTH) - 1)
+BSG_STAT_TG_ID_MASK = ((1 << BSG_STAT_TG_ID_WIDTH) - 1)
+BSG_STAT_X_MASK     = ((1 << BSG_STAT_X_WIDTH) - 1)
+BSG_STAT_Y_MASK     = ((1 << BSG_STAT_Y_WIDTH) - 1)
+BSG_STAT_TYPE_STAT  = 0
+BSG_STAT_TYPE_START = 1
+BSG_STAT_TYPE_END   = 2
```

Comment:
These should be class variables of CudaStatTag

Reply:
Stole your code, thanks! :D